service/src/main/java/tds/session/repositories/ExternRepository.java

```diff
@@ -0,0 +1,14 @@
+package tds.session.repositories;
+
+import java.util.Optional;
+
+import tds.session.Extern;
+
+public interface ExternRepository {
+    /**
+     * Fetches the extern by given client name
+     * @param clientName client name for extern
+     * @return optional containing {@link tds.session.Extern Extern} otherwise empty Optional
+     */
+    Optional<Extern> getExternByClientName(String clientName);
```

Comment:
We've been using Optional so that we can return nothing and let the consumer decide if it is an exceptional case or not.  This might be a case where we say it is really exceptional.  If there is no Extern data do we know that there are such big problems that things won't work anyways and we should just throw?  I like the consistency of not throwing, but think it's worth thinking about since I think this truly is exceptional and means there are big problems.